### PR TITLE
Small changes in detection arch solaris and AIX

### PR DESCRIPTION
**Pull request by lasote**
Changelog: omit
Docs: omit

Related to https://github.com/conan-io/conan/pull/5630
@datalogics-robb  please take a look

Extracted duplicated code to OSInfo and incorporated new Solaris arch detection to the Conan default profile detection.

### CODE REVIEW

File: conans/client/conf/detect.py [hunk 1]
```diff
@@ -194,14 +194,8 @@ def _detect_os_arch(result, output):
...
-        elif the_os == 'AIX':
-            processor = platform.processor()
-            if "powerpc" in processor:
-                kernel_bitness = OSInfo().get_aix_conf("KERNEL_BITMODE")
-                if kernel_bitness:
-                    arch = "ppc64" if kernel_bitness == "64" else "ppc32"
-            elif "rs6000" in processor:
-                arch = "ppc32"
+        elif OSInfo().is_aix:
+            arch = OSInfo.get_aix_architecture()
```
Comment (jgsogo) [inline conans/client/conf/detect.py]: Here you are overriding the value of `arch` to `None` if the processor isn't `powerpc` or `rs6000`

File: conans/client/tools/oss.py [hunk 2]
```diff
@@ -48,17 +48,13 @@ def detected_os():
...
-        if "sparc" in processor:
-            return "sparcv9" if kernel_bitness == "64bit" else "sparc"
-        elif "i386" in processor:
-            return "x86_64" if kernel_bitness == "64bit" else "x86"
- 
-    if "ppc64le" in machine:
+    if os_info.is_solaris:
+        return OSInfo.get_solaris_architecture()
+    elif os_info.is_aix:
+        return OSInfo.get_aix_architecture()
```
Comment (jgsogo) [inline conans/client/tools/oss.py]: Moving this condition here is not just a refactor. Probably this is the right place for this condition.

File: conans/client/conf/detect.py (hunk 1 above)
Comment (lasote) [reply conans/client/conf/detect.py]: We can live with that change.

File: conans/client/tools/oss.py (hunk 2 above)
Comment (lasote) [reply conans/client/tools/oss.py]: I know, I changed the title of the PR. I took it into account and I think this is ok and more clear to move it here.

Comment (lasote) [reply conans/client/tools/oss.py]: The changelog is still irrelevant for the users. The slightly changed behavior shouldn't affect anyone.

File: conans/client/tools/oss.py [hunk 3]
```diff
@@ -292,6 +284,29 @@ def get_osx_version_name(version):
...
+    def get_aix_architecture():
+        processor = platform.processor()
+        if "powerpc" in processor:
+            kernel_bitness = OSInfo().get_aix_conf("KERNEL_BITMODE")
+            if kernel_bitness:
+                return "ppc64" if kernel_bitness == "64" else "ppc32"
+        elif "rs6000" in processor:
+            return "ppc32"
+
+        return None
```
Comment (SSE4) [inline conans/client/tools/oss.py]: I think this return is not-needed? e.g. `get_solaris_architecture` doesn't have it

File: conans/client/tools/oss.py [hunk 4]
```diff
@@ -48,16 +48,17 @@ def detected_os():
 def detected_architecture():
     # FIXME: Very weak check but not very common to run conan in other architectures
     machine = platform.machine()
+    os_info = OSInfo()
```
Comment (SSE4) [inline conans/client/tools/oss.py]: if methods are static, we don't need an instance, right?

Comment (jgsogo) [reply conans/client/tools/oss.py]: Some members like `is_solaris` are initialized in the ctor; so as it is implemented right now, we need an instance

File: conans/client/tools/oss.py (hunk 3 above)
Comment (lasote) [reply conans/client/tools/oss.py]: removed

File: conans/client/tools/oss.py (hunk 4 above)
Comment (SSE4) [reply conans/client/tools/oss.py]: oh I see now

**Pull request merged**